src/relay.js

```diff
@@ -19,42 +19,61 @@ import {
 import _ from 'lodash';
 import simplifyAST from './simplifyAST';
 
-class NodeTypeMapper {
-  constructor(sequelize) {
-    this.models = Object.keys(sequelize.models);
-    this.models.forEach(model => {
-      this[model] = null;
-    });
+export class NodeTypeMapper {
+  constructor() {
+    this.map = { };
   }
 
   mapTypes(types) {
-    Object.keys(types).forEach(type => {
-      this[type] = types[type];
+    Object.entries(types).forEach(([k, v]) => {
+      this.map[k] = v.type
+        ? v
+        : { type: v };
     });
   }
+
+  item(type) {
+    return this.map[type];
+  }
 }
 
 export function idFetcher(sequelize, nodeTypeMapper) {
-  return globalId => {
-    let {type, id} = fromGlobalId(globalId);
-    const models = Object.keys(sequelize.models);
-    if (models.some(model => model === type)) {
-      return sequelize.models[type].findById(id);
+  return async globalId => {
+    const {type, id} = fromGlobalId(globalId);
+
+    const nodeType = nodeTypeMapper.item(type);
+    if (nodeType && typeof nodeType.resolve === 'function') {
+      const resOrPromise = nodeType.resolve(globalId);
+      const res = resOrPromise.then ? await resOrPromise : resOrPromise;
```

Comment:
It might be cleaner to simply have `await Promise.resolve(nodeType.resolve(globalId))`